### ISSUE TITLE
Issue with comparative augustus config file in CAT

### ISSUE THREAD

**Issue by ifiddes** [Gaius-Augustus/Augustus#220]
Hello,

A CAT user (@francicco) just had an issue that I am able to reproduce related to the config file that I don't understand. It seems to be complaining that `Etal` is not a valid species identifier, but it is present in both the MAF file and the guide tree. I have attached the files and the command line that can reproduce the issue. The error occurs in the config parsing step, so the hints database and other files are not needed to reproduce.

Command line:

```
courtyard Mon Nov 23 08:47 francesco $augustus --extrinsicCfgFile=cfg --species=heliconius_melpomene1 --treefile=tree --alnfile=aln.maf --speciesfilenames=fofn --softmasking=1 --exoncands=1 --alternatives-from-evidence=0 --/CompPred/logreg=on --printOEs=1 --/CompPred/outdir=$PWD
-------Speciesnames:--------
reading in file names for species from fofn
# Sources of extrinsic information: M RM E W PB
# Setting acceptor splice site local malus: 0.1
# Setting donor splice site local malus: 0.1
# Setting exon local malus: 0.985
# Setting CDSpart local malus: 0.985
# Setting UTRpart local malus: 0.985
extrinsic group 1:Etal is not a valid species identifier.
Could not read in file with the configuration of hints: cfg

augustus: ERROR
	Etal is not a valid species identifier.
```

Files are attached.
[francesco.tar.gz](https://github.com/Gaius-Augustus/Augustus/files/5584501/francesco.tar.gz)

**Comment by francicco** [comment]
Any update for this one? 

I just tested again CAT and Augustus-cgp and both work:

```
===== Luigi Execution Summary =====

Scheduled 195 tasks of which:
* 1 complete ones were encountered:
    - 1 ExternalReferenceFiles(...)
* 194 ran successfully:
    - 7 AlignTranscriptDriverTask(...)
    - 1 AlignTranscripts(...)
    - 1 AssemblyHub(...)
    - 1 Augustus(...)
    - 1 AugustusCgp(...)
    ...

This progress looks :) because there were no failed tasks or missing dependencies

===== Luigi Execution Summary =====
```

```
[tk19812@bc4login3 cgp]$ augustus --species=human --treefile=tree.nwk --alnfile=aln.maf --speciesfilenames=genomes.tbl --/CompPred/outdir=out
-------Speciesnames:--------
species 0	hg19
species 1	mm9
species 2	bosTau4
species 3	galGal3
reading in file names for species from genomes.tbl
reading in bosTau4.chr1
reading in galGal3.chr1
reading in hg19.chr21
reading in mm9.chr16
# No extrinsic information given.
Warning: The option 'alternatives-from-evidence' is only available in single species mode. Turned it off.
Rerun with --alternatives-from-evidence=0 to remove this warning.
# human version. Using default transition matrix.
# setting temperature to 3 (for sampling)
number of nodes: 1
number of signatures=0. First 10 signatures are:
numAli	sumCumFragLen	sumAliLen	depth	color	signature
found 0 paths
Have 0 collisions.
MILP objective=0
removing alignments 
0 geneRanges=alignments (0%) and 0 / 58758 = 0% of aligned bases and 0 / 106809 = 0% of aligment-range-covered bases were deleted because of too much redundancy.
findGeneRanges reduced the number of aligments from 313 to 1 (to 0.319%).
processing next gene range:
hg19.chr21	+	0	32084
mm9.chr16	+	0	25967
bosTau4.chr1	+	0	30874
galGal3.chr1	+	0	22282
Found 2566 ECs on species hg19
Found 2217 ECs on species mm9
Found 2643 ECs on species bosTau4
Found 2088 ECs on species galGal3
OEs in this gene range must have at least 2 ECs
Found 141 ortho exons
building Graph for hg19
building Graph for mm9
building Graph for bosTau4
building Graph for galGal3
computing omega for each ortho exon.
--- processing ortho exons on forward strand ---
generating codon alignment
Merge processing: Traverse alignment left to right
--- processing ortho exons on reverse strand ---
generating codon alignment
Merge processing: Traverse alignment left to right
compute omegas done
dual decomposition on gene Range 1
round	iter	step_size	primal	dual	#inconsistencies
0	0	0	96.42318205	137.9112869	4
0	1	10.37202622	127.1016117	168.5897175	34
0	2	1.220238405	127.1016026	149.0658933	16
0	3	1.372767597	127.1016041	127.1016136	16
0	4	1.192093046e-07	127.1016017	127.1016093	16
dual decomposition reduced initial duality gap of 30.67842727 to 0 (to 0%)
# total time: 11.7s
# command line:
# augustus --species=human --treefile=tree.nwk --alnfile=aln.maf --speciesfilenames=genomes.tbl --/CompPred/outdir=out
```

Please Help!
F

**Comment by francicco** [comment]
Sorry. Any update?
Best,
F

**Comment by diekhans** [comment]
first level of confusion is that "is not a valid species identifier."  message from randseqaccess.cc
  throw ProjectError(skey + " is not a valid species identifier.");

doesn't really tell the user the cause of the problem, which is the species identifier is not found in the tree.  It would be great to have this more explicit.

**Comment by diekhans** [comment]
Next problem is that  --speciesfilenames=fofn points to a file
that is more than just the species name (also includes fastas).

This results in PhyloTree::PhyloTree(string filename) not finding any the species subset in the keep list,  Resulting in a empty tree.

A this point, it should generate an error indicating what happened, instead  just continues until it tries to use the tree.

So I thinks this is GIGO from CAT, with AUGUSTUS not doing good error checking.

**Comment by diekhans** [comment]
Turns out there was a error that was suppose to check for all species being removed, but it was checking the wrong value.

**Comment by diekhans** [comment]
I couldn't spend all that time tracking down the problem and not just fix it.

now, back in CATs lap

**Comment by diekhans** [comment]
Nope, this is an AUGUSTUS problem.  If the --speciesfilenames file has lines longer than 255 characters, it failed to read the file.

This is now fixed in #250

**Comment by MarioStanke** [comment]
I merged your pull-request, Mark. That indeed was a bug that can explain the fail. 
Thanks a lot!
